core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/CaseSensitivitySpec.scala

```diff
@@ -463,6 +471,205 @@ class CaseSensitivitySpec extends AnyFreeSpec with Matchers with TestDriverCompo
+          // save case insensitive attribute data
+          runAndWait(entityQuery.save(testWorkspace.workspace, caseInsensitiveAttributeData))
+
+          // delete column all entities
+          val columnsToDelete = Set(attributeNameToDelete)
+          runAndWait(
+            entityAttributeShardQuery(testWorkspace.workspace).deleteAttributes(testWorkspace.workspace, "cat", columnsToDelete)
+          )
+
+          // get all attributes to verify deletion
+          val allAttributes = runAndWait(entityAttributeShardQuery(testWorkspace.workspace).result)
+            .map(attr => toDelimitedName(AttributeName(attr.namespace, attr.name)))
+
+          // verify an attribute is deleted
+          allAttributes.size shouldBe exemplarAttributeNames.size - 1
+          allAttributes shouldNot contain (toDelimitedName(attributeNameToDelete))
+
+          // verify the correct attributes remain
+          val remainingAttributeNames = exemplarAttributeNames.toSet - toDelimitedName(attributeNameToDelete)
+          allAttributes should contain allElementsOf remainingAttributeNames
+        }
+      }
+
+      "should get all attribute names when getting an entity" in withTestDataServices { _ =>
+        // save case insensitive attribute data
+        runAndWait(entityQuery.save(testWorkspace.workspace, caseInsensitiveAttributeData))
+
+        // get provider
+        val provider = new LocalEntityProvider(testWorkspace.workspace, slickDataSource, true, "metricsBaseName")
+
+        // get our entity
+        val entity = provider.getEntity("cat", "005").futureValue // Entity
+
+        // make sure all attributes are created
+        exemplarAttributeNames.size should equal(entity.attributes.size)
+        entity.attributes.keys.map(toDelimitedName) should contain allElementsOf exemplarAttributeNames
+      }
+
+      "should update attributes correctly when updating an entity" in withTestDataServices { services =>
+        exemplarAttributesMap.keys.foreach { attributeToUpdate =>
+          // first save entity (all attribute values are "bar"
+          services.entityService.createEntity(testWorkspace.wsName, caseInsensitiveAttributeData.head).futureValue
+
+          // now update one attribute value
+          val attributeUpdates = Seq(AddUpdateAttribute(attributeToUpdate, AttributeString("new-value")))
+          val updatedEntity = services.entityService.updateEntity(testWorkspace.wsName, "cat", "005", attributeUpdates).futureValue
+
+          // make sure all attributes are returned, and only the updated attribute is updated
+          updatedEntity.attributes.size shouldBe exemplarAttributeNames.size
+          updatedEntity.attributes.foreach { attributeToCheck =>
+            if (attributeToCheck._1 == attributeToUpdate)
+              attributeToCheck._2 shouldBe AttributeString("new-value")
+            else
+              attributeToCheck._2 shouldBe AttributeString("bar")
+          }
```

Comment:
Scala idioms: you should be able to use a `case` partial function here to ease tuple handling and avoid the `._1` and `._2`:
```scala
          updatedEntity.attributes.foreach {
            case (attributeName, attributeValue) =>
              if (attributeName == attributeToUpdate)
                attributeValue shouldBe AttributeString("new-value")
              else
                attributeValue shouldBe AttributeString("bar")
          }
```
no need to change the PR, this is just for your knowledge